tests/Doctrine/Tests/DBAL/Functional/MasterSlaveConnectionTest.php

```diff
@@ -139,4 +139,19 @@ public function testMasterSlaveConnectionCloseAndReconnect()
         $conn->connect('master');
         $this->assertTrue($conn->isConnectedToMaster());
     }
+
+    public function testKeepSlaveNoMasterOnExecuteQueryAfterWriteOperation()
+    {
+        $conn = $this->createMasterSlaveConnection(true);
+        $conn->insert('master_slave_table', array('test_int' => 30));
+
+        $this->assertTrue($conn->isConnectedToMaster());
+
+        $sql = "SELECT count(*) as num FROM master_slave_table";
```

Comment:
Is this a different case https://github.com/doctrine/dbal/blob/ec8fc16604915110eea8046e3ea594c133aee880/tests/Doctrine/Tests/DBAL/Functional/MasterSlaveConnectionTest.php#L67 ?